packages/x-components/src/components/result/base-result-image.vue

```diff
@@ -98,8 +153,21 @@
      * @internal
      */
     protected get imageSrc(): string {
-      const image = this.result.images?.find(image => !this.failedImages.includes(image));
-      return image ?? '';
+      return this.loadedImages[
+        !this.showNextImageOnHover || !this.isHovering ? 0 : this.loadedImages.length - 1
```

Comment:
I don't see the point on using `this.loadedImages.length - 1` and not just `1`, but probably I'm missing something I think.

Reply:
Yep, it is `this.loadedImages.length - 1` for two reasons:
- The second image loads when the user hovers the picture the first time, so accessing the index 1 before the second image is loaded would cause an undefined index.
- The second image can fail to load. In that case, the first image should remain visible.